aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -3107,10 +3107,10 @@ Tensor linalg_tensorinv(const Tensor& self, int64_t ind) {
 
   // If the reshaped self is not invertible catch this error
   Tensor result, info;
-  std::tie(result, info) = at::linalg_inv_ex(self.reshape({prod_ind_end, prod_ind_end}), /*check_errors=*/false);
+  std::tie(result, info) = at::linalg_inv_ex(self.reshape_symint({prod_ind_end, prod_ind_end}), /*check_errors=*/false);
```

Comment:
Nit: since we are on C++17, can std::tie be changed to structured bindings?
```suggestion
  auto [result, info] = at::linalg_inv_ex(self.reshape_symint({prod_ind_end, prod_ind_end}), /*check_errors=*/false);
```